stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -35,6 +35,24 @@ public sealed class Result<out A : Any> {
     public val isFailure: Boolean
         inline get() = this is Failure
 
+    /**
+     * Returns the encapsulated value if this instance represents [Success] [Result.isSuccess] or `null`
+     * if it is [Failure] [Result.isFailure].
+     */
+    public fun getOrNull(): A? = when (this) {
+        is Success -> value
+        is Failure -> null
+    }
```

Comment:
What do you think about adding `getOrThrow` as well, which is one of the popular extensions of the `Result` class?

Reply:
that is a good idea. Let me add it